### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by uga-rosa**
Use flatMap instead of for nesting.

This is not a bug fix, so you can close it if it is not to your taste.

### CODE REVIEW

File: denops/@ddu-sources/help.ts [hunk 1]
```diff
@@ -55,7 +50,7 @@ export class Source extends BaseSource<Params> {
               const root = dirname(f);
               lines.map((line) => {
                 const seg = line.split("\t");
-                if (seg.length < 2) return;
+                if (seg.length < 3) return;
```
Comment (uga-rosa) [inline denops/@ddu-sources/help.ts:53]: With 2, the pattern can be undefined.

File: denops/@ddu-sources/help.ts [hunk 2]
```diff
@@ -70,30 +65,28 @@ export class Source extends BaseSource<Params> {
           );
           await Promise.all(fileReadPromise);
 
-          const items: Item<ActionData>[] = [];
-          Object.keys(tagsMap).map((tag) => {
-            const info = tagsMap[tag];
-            if (args.sourceParams.style == "minimal" || info.length < 2) {
-              items.push({
+          const items = Object.entries(tagsMap).flatMap(([tag, infos]) => {
+            if (args.sourceParams.style === "minimal" || infos.length < 2) {
```
Comment (uga-rosa) [inline denops/@ddu-sources/help.ts:69]: We should use the strict equality (===) operator.

File: denops/@ddu-sources/help.ts [hunk 3]
```diff
@@ -55,7 +50,7 @@ export class Source extends BaseSource<Params> {
               const root = dirname(f);
               lines.map((line) => {
                 const seg = line.split("\t");
-                if (seg.length < 2) return;
+                if (seg.length < 3) return;
                 const [tag, path, pattern] = seg;
```
Comment (uga-rosa) [inline denops/@ddu-sources/help.ts:54]: With 2, `pattern` can be undefined.

**Comment by matsui54** [comment]
This PR looks good. Is there something wrong?

**Comment by uga-rosa** [comment]
I have many other changes I would like to make for my own use, and I am using the forking version with many changes.
Shall I reopen this?

**Comment by matsui54** [comment]
OK! Please let me know if it's ready to merge.

**Comment by uga-rosa** [comment]
Ok. You can merge this PR.

**Comment by uga-rosa** [comment]
🤔

**Review by matsui54 — CHANGES_REQUESTED** [review]

**Comment by matsui54** [comment]
Sorry, I didn't realize my review was pending🙇

**Comment by uga-rosa** [comment]
I see.
Fixed.

**Pull request merged**